pkgs/applications/science/math/matlab/fetcher.nix

```
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenvNoCC,
+  mpm,
+}:
+
+{
+  release,
+  update ? 0,
+  extra-products ? [ ],
+  hash,
+}:
+
+let
+  version = "${release}U${toString update}";
+  products = [ "MATLAB" ] ++ extra-products;
+in
+stdenvNoCC.mkDerivation {
+  pname = "matlab-download";
+  inherit version;
+
+  dontUnpack = true;
+
+  installPhase = ''
+    # As mpm is bubblewrapped, it cannot write to $out directly
+    ${mpm}/bin/mpm download --release ${version} --platforms glnxa64 --destination matlab --products ${lib.escapeShellArgs products}
+    mv matlab $out
+  '';
+
+  dontFixup = true;
+
+  # The MATLAB installation isn't 100% deterministic due to how the classpath.txt file is generated.
+  # But mpm download does appear to be deterministic. So create a fixed output derivation from the
+  # downloaded files, and then run the installer.
+  outputHashAlgo = "sha256";
```

The `hash` attribute is meant to be algorithm-agnostic wherever possible. That's the purpose of the tree-wide migration from `sha256 =` to `hash =` in Nixpkgs.

```suggestion
  outputHashAlgo = lib.optionalString (hash == "") "sha256";
```